prover/src/constraints/evaluator.rs

```diff
@@ -335,4 +335,37 @@ impl<'a, A: Air, E: FieldElement<BaseField = A::BaseField>> ConstraintEvaluator<
     fn num_aux_transition_constraints(&self) -> usize {
         self.transition_constraints.num_aux_constraints()
     }
+
+    // HELPERS
+    // --------------------------------------------------------------------------------------------
+
+    // This is a modified version of the TransitionConstraintGroup::merge_evaluations method. This
+    // implementation uses pre-computed of powers of generator of the constraint evaluation domain.
+    // This allows up to substitute the exponentiations needed for the degree adjustment factor 
+    // with lookups. 
+    pub fn merge_evaluations_optimized<F>(
```

Comment:
Rather than creating a separate method for this, I'd probably change the signature of `TransitionConstraintGroup::merge_evaluations` to something like this:
```Rust
pub fn merge_evaluations<B, F>(&self, evaluations: &[F], xp: B) -> E
```
So, instead of `x` we'd be passing `xp` into this function and how `xp` is computed would be different on the prover and verifier sides. (i.e., on the prover side we won't use `exp`).